Replace configuration variables with configuration (#2144)
balena-io/docs: pages/learn/manage/configuration.md

BALENA_SUPERVISOR_VPN_CONTROL | true | Enable / Disable VPN | v1.3.0
BALENA_SUPERVISOR_INSTANT_UPDATE_TRIGGER | true | Enable / Disable instant triggering of updates when a new release is deployed. If set to false, the device will ignore the notification that is triggered when the device's target state has changed. In this case, the device will rely on polling to apply updates. Note: You can spread out updates on devices if you disable instant updates and specify a different poll interval for each device in your fleet. This avoids overloading local networks if they are all at one location. | v9.13.0

In addition to these values, there may be some device-type specific configuration that can be set. For example, these are a few of the values that apply to Raspberry Pi devices, corresponding to the contents of the [Raspberry Pi `config.txt` file](https://www.raspberrypi.org/documentation/configuration/config-txt/README.md):

{{> "general/config-variables-pi" }}

You can find more information on updating `config.txt` through the configuration tab in our [Advanced Boot Configuration Guide][boot-config-guide].

> ## Managing fleet level configuration

Reviewer: ```suggestion
## Fleet configuration management
```

> Variables defined on the fleet wide configuration control the behavior of any devices running in that fleet unless they are redefined with a device level variable of the same name.

Reviewer: ```suggestion
Configuration defined at the fleet level controls the behavior of any devices running in that fleet unless it is overridden with device configuration of the same name.
```

> If you want to change the default value for one of the prepopulated configuration, make sure you are in the *Configuration* tab on the fleet level, then click *activate* for the variable you wish to define:

Reviewer: ```suggestion
If you want to change the default value for any of the pre-populated configuration, make sure you are in the *Configuration* tab on the fleet, then click *activate* for the variable you wish to define:
```

To define a custom fleet wide configuration, scroll past the automatically populated variables, then click the *Add custom variable* button in the lower-right corner:

<img alt="Add custom fleet variable" src="/img/configuration/add_fleet_configuration.png" width="100%">

A dialog pops up giving you the option to define a name and value (remember, all config variable names must begin with `{{ $names.company.allCaps}}_` or `RESIN_`). Click *Add* to apply:

<img alt="Add custom fleet configuration" src="/img/configuration/variable_editor_config.png" width="60%">

If you have already defined configuration at the device level, they will appear in a list below your fleet wide configuration. You can easily apply a device value to the entire fleet by clicking *Define fleet-wide*:

<img alt="Define device level configuration fleet-wide" src="/img/configuration/define_app_wide.png" width="100%">

> ## Managing device level configuration

Reviewer: ```suggestion
## Device configuration management
```

> Device specific configuration define the behavior of a single device. If both the fleet and the device have configuration of the same name, the code on the device will use the value of the device level configuration. In other words, device level configuration variables redefine (or override) fleet wide configuration.

Reviewer: ```suggestion
Device configuration defines the behavior of a single device. If both the fleet and the device have configuration of the same name, the code on the device will use the value given in the device configuration. In other words, device level configuration redefines (or overrides) fleet wide configuration.
```

<img alt="Define device level configuration fleet-wide" src="/img/configuration/define_app_wide.png" width="100%">

## Device configuration management

Device configuration defines the behavior of a single device. If both the fleet and the device have configuration of the same name, the code on the device will use the value given in the device configuration. In other words, device level configuration redefines (or overrides) fleet wide configuration.

The device level configuration list includes the pre-populated default values. Variables that can be enabled or disabled can be easily switched with the provided toggles. Other values can be edited by clicking the small edit (pencil) icon.

<img alt="Device configuration" src="/img/configuration/device_config_variables.png" width="100%">

Clicking the edit icon will pop up a small dialog for editing the value:

<img alt="Edit device configuration" src="/img/configuration/edit_device_config.png" width="60%">

To remove the device level configuration, and reset it to its default value, click the delete (trash can) icon.

### Adding custom configuration

> Custom configuration section can be used to modify configuration options beyond the ones pre-populated for your device using the balenaCloud dashboard. Examples include, [modifying config.txt using configuration variables][boot-config-guide] for Raspberry Pi devices.

Reviewer: ```suggestion
The custom configuration section can be used to modify configuration options beyond the ones pre-populated for your device using the balenaCloud dashboard. Examples include [modifying config.txt using configuration variables][boot-config-guide] for Raspberry Pi devices.
```
